Implement table() trait

24seconds opened the pull request:
I watched the video in Issue #36 and it seems interesting. So I made `TableIteratorExt` trait to support `table()` function. 
(If the comment are needed let me add it. If the trait name is not good let me change it)

```rust
let people: Vec<Person> = ...
let table = people.table()
                  .with(Style::psql());

```
The `TableIteratorExt` applies to the type `U` which is bound to `IntoIterator<Item = T>` type. (T is bound to `Tabled` type)

I don't know which tests to be added, so I simply added one test to `table_test.rs`

zhiburt commented:
Hi @24seconds 

That looks amazing.

Might you add a doc comment to `TableIteratorExt`.
And some notion about it in `README.md`?

I can do it myself if you wish.

24seconds commented:
@zhiburt Thanks for the reply. I added comment to `TableIteratorExt`. Could you write notion about in `README.md`? I don't know which part would be proper. Of course, you can modify the comment I added :D.

zhiburt commented:
Thank you for your contribution.
I'll merge it today.

Marry Christmas and Happy New Year
